lib/paperclip/storage/fog.rb

```diff
@@ -141,17 +152,18 @@ def public_url(style = default_style)
 
       def expiring_url(time = (Time.now + 3600), style_name = default_style)
         time = convert_time(time)
-        if path(style_name) && directory.files.respond_to?(:get_http_url)
-          expiring_url = directory.files.get_http_url(path(style_name), time)
-
+        url_method = url_protocol == 'https' ? :get_https_url : :get_http_url
+        if path(style_name) && directory.files.respond_to?(url_method)
+          expiring_url = directory.files.send(url_method, path(style_name), time)
           if @options[:fog_host]
-            expiring_url.gsub!(/#{host_name_for_directory}/, dynamic_fog_host_for_style(style_name))
+            fog_host_without_protocol = dynamic_fog_host_for_style(style_name).gsub(/\Ahttps?:\/\//, '')
```

Comment:
Line is too long. [104/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.